src/OpenLoco/src/Ui/Windows/Construction/ConstructionTab.cpp

```diff
@@ -2543,18 +2552,67 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
             return;
         }
 
+        if (_isDragging)
+        {
+            mapInvalidateMapSelectionTiles();
+            removeConstructionGhosts();
+            return;
+        }
+
         if (_cState->trackType & (1 << 7))
         {
-            onToolUpdateTrack(x, y, getRoadPieceId, tryMakeRoadJunctionAtLoc, TrackData::getRoadPiece, getRoadPlacementArgs, placeRoadGhost);
+            onToolUpdateSingle(x, y, getRoadPieceId, tryMakeRoadJunctionAtLoc, TrackData::getRoadPiece, getRoadPlacementArgs, placeRoadGhost);
         }
         else
         {
-            onToolUpdateTrack(x, y, getTrackPieceId, tryMakeTrackJunctionAtLoc, TrackData::getTrackPiece, getTrackPlacementArgs, placeTrackGhost);
+            onToolUpdateSingle(x, y, getTrackPieceId, tryMakeTrackJunctionAtLoc, TrackData::getTrackPiece, getTrackPlacementArgs, placeTrackGhost);
+        }
+    }
+
+    static void onToolDown([[maybe_unused]] Window& self, [[maybe_unused]] const WidgetIndex_t widgetIndex, [[maybe_unused]] const WidgetId id, const int16_t x, const int16_t y)
+    {
+        auto res = ViewportInteraction::getMapCoordinatesFromPos(x, y, ~(ViewportInteraction::InteractionItemFlags::surface | ViewportInteraction::InteractionItemFlags::water));
+        auto& interaction = res.first;
+        if (interaction.type == ViewportInteraction::InteractionItem::noInteraction)
+        {
+            return;
+        }
+
+        _toolPosInitial = World::toTileSpace(interaction.pos);
+        _isDragging = false;
+    }
+
+    static void onToolDrag([[maybe_unused]] Window& self, [[maybe_unused]] const WidgetIndex_t widgetIndex, [[maybe_unused]] const WidgetId id, [[maybe_unused]] const int16_t x, [[maybe_unused]] const int16_t y)
+    {
+        mapInvalidateSelectionRect();
+        removeConstructionGhosts();
+
+        auto res = ViewportInteraction::getMapCoordinatesFromPos(x, y, ~(ViewportInteraction::InteractionItemFlags::surface | ViewportInteraction::InteractionItemFlags::water));
+        auto& interaction = res.first;
+        if (interaction.type == ViewportInteraction::InteractionItem::noInteraction)
+        {
+            return;
+        }
+
+        _toolPosDrag = World::toTileSpace(interaction.pos);
+        _isDragging = _toolPosInitial != _toolPosDrag;
+        if (!_isDragging)
+        {
+            return;
         }
+
+        auto posA = World::toWorldSpace({ std::min(_toolPosInitial.x, _toolPosDrag.x), std::min(_toolPosInitial.y, _toolPosDrag.y) });
+        auto posB = World::toWorldSpace({ std::max(_toolPosInitial.x, _toolPosDrag.x), std::max(_toolPosInitial.y, _toolPosDrag.y) });
```

Comment:
If we have a good way to just get the minimum of the two vectors, I'd love to hear it. I didn't see one in `Math::Vector`, nor does `operator<=>` appear to be defined.

Reply:
In Unity they have Vector2/3.Min and Max which does that but I somewhat fear we'll just have a clash with those min/max macros, I'm not sure if we define NOMINMAX project wide.

Reply:
Pretty much all game engines with vectors have min/max; I would just add a `Vector2::min()`

Reply:
> I'm not sure if we define NOMINMAX project wide.

Looks like we don't. We only explicitly define it in these four source files:
```
src/Interop/src/Interop.cpp
src/OpenLoco/src/OpenLoco.cpp
src/OpenLoco/src/Ui.cpp
src/Platform/src/Platform.Windows.cpp
```

Reply:
> > I'm not sure if we define NOMINMAX project wide.
> 
> Looks like we don't. We only explicitly define it in these four source files:
> 
> ```
> src/Interop/src/Interop.cpp
> src/OpenLoco/src/OpenLoco.cpp
> src/OpenLoco/src/Ui.cpp
> src/Platform/src/Platform.Windows.cpp
> ```
> 
> Perhaps we should just define it globally in `CMakeLists.txt`?

Tthis is probably isolated to Windows, most likely windows.h, I don't quite remember where that comes from. I think as long we don't include the problematic header it should be generally fine and it seems that we explicitly define them so the chances of breakage is quite low.